### DISCUSSION TITLE
Wifi Buzzer

### DISCUSSION THREAD

**Discussion by laguinch** [photobooth-app/photobooth-app#227] [Q&A] [answered]
Hy is it possible to use this wifi buzzer https://github.com/PhotoboothProject/photobooth-ino/blob/master/esp8266-wifi-buzzer.ino whith your photobooth ?

**Comment by mgineer85** [comment, marked as answer]
Sure, there is an open REST-API that can be used: 
https://photobooth-app.org/reference/api/

You find a local interactive api client to test in your photobooth-app -> admin center -> help -> Photobooth REST-API -> [API Schema](http://localhost:8000/api/doc)

Additionally this buzzer is also available:
https://github.com/photobooth-app/photobooth-buzzer

**Reply by mgineer85** [reply]
So in your ino you will need this line 
```
http.begin(client, "http://xx.xx.xx.xx:14711/commands/start-collage");
```

to be something like this for simple picture
```
http.begin(client, "http://xx.xx.xx.xx:8000/api/processing/chose/1pic");
```
xx.xxx ... has to be replaced by the IP of the device of course.

Or you choose other commands:
![image](https://github.com/photobooth-app/photobooth-app/assets/3868450/043f2543-cd9c-4601-a42c-37e0d9c12ac5)

**Reply by mgineer85** [reply]
The api endpoint changed in version 2 compared to 1.2.1, so please check your actual installation rest api as described above.

**Reply by laguinch** [reply]
Thx for your answer. I have a D1 in stock that’s why i want to use this buzzer. I update the photobooth to 2.0 version. I try to make it run because I have thé Same pb with my wired buzzer on the 2.0 as with the 1.2.1

**Reply by mgineer85** [reply]
Hey, I guess you are from france? Maybe you can help too improve french translation and approve?
https://github.com/photobooth-app/photobooth-app/blob/main/CONTRIBUTING.md#help-translate-the-app

**Reply by laguinch** [reply]
Hy of course I could help translating in French. 
Ya ich kann in französisch übersetzen

**Reply by laguinch** [reply]
I do it

**Reply by mgineer85** [reply]
Awesome, thank you!
Let me know if you have problems or you finished so I update the sources.

**Reply by laguinch** [reply]
I finish the translate

**Reply by laguinch** [reply]
Hy i have a problem with the buzzer, I use esp32-wroom-32 chip : `Processing esp32dev (platform: espressif32; board: esp32dev; framework: arduino)
-------------------------------------------------------------------------------------------------------------------------------------
Library Manager: Installing t-vk/ESP32 BLE Keyboard @ ^0.3.2
Unpacking  [####################################]  100%
Library Manager: ESP32 BLE Keyboard@0.3.2 has been installed!
Library Manager: Installing lennarthennigs/Button2 @ ^2.2.4
Unpacking  [####################################]  100%
Library Manager: Button2@2.3.2 has been installed!
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32dev.html
PLATFORM: Espressif 32 (6.6.0) > Espressif ESP32 Dev Module
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES: 
 - framework-arduinoespressif32 @ 3.20014.231204 (2.0.14) 
 - tool-esptoolpy @ 1.40501.0 (4.5.1) 
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 35 compatible libraries
Scanning dependencies...
Dependency Graph
|-- ESP32 BLE Keyboard @ 0.3.2
|-- Button2 @ 2.3.2
|-- ESP32 BLE Arduino @ 2.0.0
Building in release mode
Compiling .pio/build/esp32dev/src/main.cpp.o
Building .pio/build/esp32dev/bootloader.bin
Generating partitions .pio/build/esp32dev/partitions.bin
esptool.py v4.5.1
Creating esp32 image...
Merged 1 ELF section
Successfully created esp32 image.
Compiling .pio/build/esp32dev/liba80/BLE/BLE2902.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLE2904.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEAddress.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEAdvertisedDevice.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEAdvertising.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEBeacon.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLECharacteristic.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLECharacteristicMap.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEClient.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEDescriptor.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEDescriptorMap.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEDevice.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEEddystoneTLM.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEEddystoneURL.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEExceptions.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEHIDDevice.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLERemoteCharacteristic.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLERemoteDescriptor.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLERemoteService.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEScan.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLESecurity.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEServer.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEService.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEServiceMap.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEUUID.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEUtils.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/BLEValue.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/FreeRTOS.cpp.o
Compiling .pio/build/esp32dev/liba80/BLE/GeneralUtils.cpp.o
Compiling .pio/build/esp32dev/lib4b2/ESP32 BLE Keyboard/BleKeyboard.cpp.o
Compiling .pio/build/esp32dev/lib23d/Button2/Button2.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/Esp.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/FirmwareMSC.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/FunctionalInterrupt.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/HWCDC.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/HardwareSerial.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/IPAddress.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/IPv6Address.cpp.o
Archiving .pio/build/esp32dev/liba80/libBLE.a
Archiving .pio/build/esp32dev/lib4b2/libESP32 BLE Keyboard.a
Archiving .pio/build/esp32dev/lib23d/libButton2.a
Indexing .pio/build/esp32dev/lib4b2/libESP32 BLE Keyboard.a
Indexing .pio/build/esp32dev/liba80/libBLE.a
Indexing .pio/build/esp32dev/lib23d/libButton2.a
Compiling .pio/build/esp32dev/FrameworkArduino/MD5Builder.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/Print.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/Stream.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/StreamString.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/Tone.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/USB.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/USBCDC.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/USBMSC.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/WMath.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/WString.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/base64.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/cbuf.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-adc.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-bt.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-cpu.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-dac.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-gpio.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-i2c-slave.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-i2c.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-ledc.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-matrix.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-misc.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-psram.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-rgb-led.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-rmt.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-sigmadelta.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-spi.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-time.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-timer.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-tinyusb.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-touch.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/esp32-hal-uart.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/firmware_msc_fat.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/libb64/cdecode.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/libb64/cencode.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/main.cpp.o
Compiling .pio/build/esp32dev/FrameworkArduino/stdlib_noniso.c.o
/Users/yann/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-uart.c: In function 'uartSetPins':
/Users/yann/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-uart.c:153:9: warning: 'return' with no value, in function returning non-void
         return;
         ^~~~~~
/Users/yann/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-uart.c:149:6: note: declared here
 bool uartSetPins(uint8_t uart_num, int8_t rxPin, int8_t txPin, int8_t ctsPin, int8_t rtsPin)
      ^~~~~~~~~~~
Compiling .pio/build/esp32dev/FrameworkArduino/wiring_pulse.c.o
Compiling .pio/build/esp32dev/FrameworkArduino/wiring_shift.c.o
Archiving .pio/build/esp32dev/libFrameworkArduino.a
Indexing .pio/build/esp32dev/libFrameworkArduino.a
Linking .pio/build/esp32dev/firmware.elf
Retrieving maximum program size .pio/build/esp32dev/firmware.elf
Checking size .pio/build/esp32dev/firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [=         ]  11.7% (used 38280 bytes from 327680 bytes)
Flash: [========  ]  84.4% (used 1105849 bytes from 1310720 bytes)
Building .pio/build/esp32dev/firmware.bin
esptool.py v4.5.1
Creating esp32 image...
Merged 26 ELF sections
Successfully created esp32 image.
=================================================== [SUCCESS] Took 16.93 seconds ===================================================` I can't detect the buzzer in bluetooth
